### ISSUE TITLE
page.widgets() generator SegFault 11

### ISSUE THREAD

**Issue by dgallitelli** [pymupdf/PyMuPDF#520]
_**Please provide all mandatory information!**_

## Describe the bug (mandatory)
While iterating on the widgets, obtained via the `page.widgets()` generator, a Segmentation Fault error is generated. When trying to access the second widget, after doing whatever operation is expected on the widget, it generates a SegFault.

## To Reproduce (mandatory)
```python
import fitz
doc = fitz.Document(<anypdf>)
for page in doc:
    for w in page.widgets():
        print(w.field_value)
        # Note if using (break) here, no Seg Fault is caused. 
```

## Expected behavior (optional)
No Seg Fault

## Screenshots (optional)
N/A

## Your configuration (mandatory)
 - MacOS 10.14.6
 - Python 3.7.3, same thing on Python 3.8 and on AWS Lambda (using a Layer on both p37 and p38)
 - PyMuPDF version: 1.17 from `pip`

For example, the output of `print(sys.version, "\n", sys.platform, "\n", fitz.__doc__)` would be sufficient (for the first two bullets).

## Additional context (optional)
This behaviour does not happen on PyMuPDF 1.16.9, while still happens on 1.16.8

**Comment by JorjMcKie** [comment]
Thanks for reorting this.
Confirming the observation. It is a new error in v1.17.0 (worked prior to that). I reproduced it on several Windows and Linux versions.
Annotations seem to work.

**Comment by JorjMcKie** [comment]
Found and solved the problem. Was caused by dropping same memory multiple times ...
Will publish this with the next version. Please let me know if there is any urgency for you.

**Comment by dgallitelli** [comment]
Hi there, thanks for checking out the bug.
I can use a Lambda Layer with version 1.16.8 (pre-built by KLayers) with no problem for now, therefore there is no urgency.

I haven't tried Annotations though, what's the difference between those and Widgets? I don't think I can use them for reading filled forms.

**Comment by JorjMcKie** [comment]
> I haven't tried Annotations though, what's the difference between those and Widgets? I don't think I can use them for reading filled forms.

No, annots are something functionally entirely different. I mentioned them, because they share most of the PDF specifications with annots, and also >80% of the code in (Py-) MuPDF.

**Comment by JorjMcKie** [comment]
Fixed by v1.17.1 which is currently being uploaded to PyPI.
